drafts/bitcartel-nu0-network-handshake/draft.rst

```diff
@@ -0,0 +1,237 @@
+            if (node->nVersion < OVERWINTER_PROTO_VERSION) {
+                vTmpEvictionCandidates.push_back(node);
+            }
+        }
+
+        // Prioritize these nodes by replacing eviction set with them
+        if (vTmpEvictionCandidates.size() > 0) {
+            vEvictionCandidates = vTmpEvictionCandidates;
+        }
+    }
+
+The existing method of disconnecting a candidate remains:
+
+    vEvictionCandidates[0]->fDisconnect = true;
+
+The existing eviction process will classify and divide eviction candidates into buckets called netgroups.  If a netgroup only has one peer, it will not be evicted.  This means at least one pre-Overwinter node will remain connected upto the activation block height, barring any network issues or a high ban score.
+
+
+Disconnecting Existing Connections
+----------------------------------
+
+At the activation block height, an Overwinter node may still remain connected to pre-Overwinter nodes.  Currently, when connecting, a node can only perform the networking handshake once, where it sends the version message before any other messages are processed.  To disconnect existing pre-Overwinter connections, ``ProcessMessage`` is modified so that once Overwinter activates, if necessary, the protocol version of an existing peer is validated when inbound messages arrive.
+
+Example code::
+
+    bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv, int64_t nTimeReceived)
+        ...
+        else if (pfrom->nVersion == 0)
+        {
+            // Must have a version message before anything else
+            Misbehaving(pfrom->GetId(), 1);
+            return false;
+        }
+        else if (strCommand == "verack")
+        {
+            ...
+        }
+
+        // Disconnect existing peer connection when:
+        // 1. Minimum peer version is less than Overwinter version
+        // 2. The version message has been received from a peer
+        // 3. The peer's version is pre-Overwinter
+        // 4. Overwinter is active
+        else if (
+            MIN_PEER_PROTO_VERSION < OVERWINTER_PROTO_VERSION &&
+            pfrom->nVersion != 0 &&
+            pfrom->nVersion < OVERWINTER_PROTO_VERSION &&
+            NetworkUpgradeActive(GetHeight(), chainparams.GetConsensus(), Consensus::UPGRADE_OVERWINTER))
+        {
+            LogPrintf("peer=%d using obsolete version %i; disconnecting\n", pfrom->id, pfrom->nVersion);
+            pfrom->PushMessage("reject", strCommand, REJECT_OBSOLETE,
+                                strprintf("Version must be %d or greater", OVERWINTER_PROTO_VERSION));
+            pfrom->fDisconnect = true;
+            return false;
+        }
+
+
+
+Deployment
+==========
+
+This proposal will be deployed with the Overwinter network upgrade.
+
+
+Backward compatibility
+======================
+
+This proposal intentionally creates what is known as a hard fork where Overwinter nodes disconnect from pre-Overwinter nodes.
```

Comment:
This proposal doesn't itself create a hard fork, as (ignoring other changes that *do* make it such) a third party could be relaying blocks between the two sides. Remove this line.